### PR TITLE
mc_pos_control: set yaw based on attitude if local-position topic not…

### PR DESCRIPTION
**Pull request by Stifael**
follow up of https://github.com/PX4/Firmware/pull/10683

It is to note that this is a HACK and attitude subscription is not meant to be part of the mc-pos-controller.
A much better solution would be to remove Stabilized from the mc_pos_control and move it into the attitude-controller. This also requires to introduce a new topic that contains thrust-setpoint and yaw-septoint fields which would be sent from the mc-pos-ctrl and received in the attitude-ctrl. 
The conversion from thrust- and yaw-setpoint to attitude would be done in the attitude controller.

### CODE REVIEW

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -440,6 +443,12 @@ MulticopterPositionControl::poll_subscriptions()
 	if (updated) {
 		orb_copy(ORB_ID(vehicle_trajectory_waypoint), _traj_wp_avoidance_sub, &_traj_wp_avoidance);
 	}
+
+	orb_check(_vehicle_att_sub, &updated);
```
Comment (dagar) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: Comment why this is being done and perhaps only bother with the orb check and copy if you're in the relevant condition (non-finite lpos yaw?).

**Comment by mhkabir** [comment]
How hard is it to move it to the attitude controller all at once instead of these incremental hacks?

**Comment by bkueng** [comment]
> How hard is it to move it to the attitude controller all at once instead of these incremental hacks?

@Stifael do you have time for that?

> This also requires to introduce a new topic that contains thrust-setpoint and yaw-septoint fields which would be sent from the mc-pos-ctrl and received in the attitude-ctrl.

Why does it require an interface change?

**Comment by Stifael** [comment]
> @Stifael do you have time for that?

Not now, that is why added that "hack".

> Why does it require an interface change?

It depends what you want. I prefer to have the mapping from thrust-to-attitude (https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L765) in the attitude controller, otherwise Stabilize would require the same mapping the attitude controller again. 
I would seek for having three kinds of inputs to the attitude controller:
- Thrust setpoint + yaw (position, altitude etc)
- Attitude setpoint (from an offboard module)
- Stick inputs

The stick inputs are required for stabilized (and of course for rate-controller, but that is a different topic). 
In stabilized, the stick inputs are first mapped to thrust setpoint + yaw and then mapped to attitude using the same Thrust-to-Attitude map.

**Comment by Stifael** [comment]
@dagar I added the comments and move the orb_check

**Pull request closed without merging**